### PR TITLE
refac: prepare circle stark

### PR DESCRIPTION
**Pull request by chokobole**
# Description

This PR is made to prepare for circle stark

### CODE REVIEW

File: vendors/circom/circomlib/json/groth16_proof.h [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 #include "circomlib/json/json_converter_forward.h"
 #include "circomlib/json/points.h"
+#include "tachyon/base/logging.h"
```
Comment (ashjeong) [inline vendors/circom/circomlib/json/groth16_proof.h:6]: how does this relate to uniting circle point to affine points?

**Review by ashjeong — COMMENTED** [review]
https://github.com/kroma-network/tachyon/pull/493/commits/3e4802c5bc149327a5f97f0d9024d6708d4d87a1?diff=split&w=0
If I'm getting this correct, `CirclePoint` was created, but since it's technically an affine point, the name and type was changed as such. The title seems to suggest that circle points was combined with affine points. Perhaps "change circle point to affine point" would be better?

File: vendors/circom/circomlib/json/groth16_proof.h (hunk 1 above)
Comment (chokobole) [reply vendors/circom/circomlib/json/groth16_proof.h:6]: This changes comes from a different change, I'll move this change into different commit

**Review by ashjeong — APPROVED** [review]
LGTM

**Review by GideokKim — APPROVED** [review]
LGTM

File: tachyon/math/base/semigroups.h
Comment (TomTaehoonKim) [inline tachyon/math/base/semigroups.h:1]: Can we remove the FIXME comment in line 300?

Comment (chokobole) [reply tachyon/math/base/semigroups.h:1]: Still there's error about this. If I modified it to `operator*()`, I faced this error.

```
bazel-out/k8-dbg/bin/tachyon/c/math/elliptic_curves/bn/bn254/g1.cc:71:35:   required from here
./tachyon/math/elliptic_curves/short_weierstrass/projective_point.h:94:34: error: request for member 'operator*' is ambiguous
   94 |     return ScalarField::Random() * Generator();
      |            ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
In file included from ./tachyon/math/base/groups.h:15,
                 from ./tachyon/math/base/ring.h:8,
                 from ./tachyon/math/base/field.h:7,
                 from ./tachyon/math/finite_fields/finite_field.h:4,
                 from ./tachyon/math/finite_fields/prime_field_base.h:24,
                 from bazel-out/k8-dbg/bin/tachyon/math/elliptic_curves/bn/bn254/fq_prime_field_x86.h:15,
                 from bazel-out/k8-dbg/bin/tachyon/math/elliptic_curves/bn/bn254/fq.h:9,
                 from bazel-out/k8-dbg/bin/tachyon/c/math/elliptic_curves/bn/bn254/fq_type_traits.h:8,
                 from bazel-out/k8-dbg/bin/tachyon/c/math/elliptic_curves/bn/bn254/g1.cc:5:
./tachyon/math/base/semigroups.h:97:18: note: candidates are: 'template<class G2, std::enable_if_t<tachyon::math::internal::SupportsMul<tachyon::math::PrimeField<tachyon::math::bn254::FrConfig>, G2, void>::value, void>* <anonymous> > constexpr auto tachyon::math::MultiplicativeSemigroup<G>::operator*(const G2&) const [with G2 = G2; std::enable_if_t<tachyon::math::internal::SupportsMul<G, G2>::value>* <anonymous> = <anonymous>; G = tachyon::math::PrimeField<tachyon::math::bn254::FrConfig>]'
   97 |   constexpr auto operator*(const G2& other) const {
      |                  ^~~~~~~~
./tachyon/math/base/semigroups.h:312:32: note:                 'template<class Scalar> constexpr auto tachyon::math::AdditiveSemigroup<G>::operator*(const Scalar&) const [with Scalar = Scalar; G = tachyon::math::PrimeField<tachyon::math::bn254::FrConfig>]'
  312 |   [[nodiscard]] constexpr auto operator*(const Scalar& scalar) const {
      |                                ^~~~~~~~
./tachyon/math/base/semigroups.h:307:32: note:                 'template<long unsigned int N> constexpr auto tachyon::math::AdditiveSemigroup<G>::operator*(const tachyon::math::BigInt<N2>&) const [with long unsigned int N = N; G = tachyon::math::PrimeField<tachyon::math::bn254::FrConfig>]'
  307 |   [[nodiscard]] constexpr auto operator*(const BigInt<N>& scalar) const {
      |                               
```

**Review by dongchangYoo — APPROVED** [review]
LGTM

**Comment by dongchangYoo** [comment]
I understand that `geometry` is related to vectors. However, it seems that concepts such as `circle` and `elliptic curves` are increasingly being integrated. Therefore, it might be a good idea to rename the package in the future or create a new, more appropriate package.

File: vendors/circom/circomlib/json/groth16_proof.h (hunk 1 above)
Comment (batzor) [reply vendors/circom/circomlib/json/groth16_proof.h:6]: @chokobole @ashjeong I think it is still unreflected.

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

**Review by batzor — APPROVED** [review]
LGTM

**Pull request merged**